### DISCUSSION TITLE
Context Length setting is completely ineffective

### DISCUSSION THREAD

**Discussion by CookSleep** [open-webui/open-webui#4983] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.16
- **Operating System:** Ubuntu 22.04
- **Browser:** Edge 128.0.2739.42

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

After modifying the Context Length to a specific value in the account settings or conversation settings, the sent requests will follow the settings.

No matter where or how the Context Length is set, it seems that the setting does not take effect. The tokens consumed by the requests, the balance, and the model's ability to remember information far exceeding the Context Length limit all reflect this.

## Description

**Bug Summary:**
Context Length setting is completely ineffective

## Reproduction Details

**Steps to Reproduce:**
1. Modify Context Length to 128000 in account settings
2. Modify Context Length to 50 in the Chat Controls of a new conversation
3. Send the model "Please remember, 47867 represents a bug" (randomly written)
4. After the model replies, send a new message that is far longer than 50 tokens (about 4000 tokens) and completely unrelated (part of the environment variable content from the Open WebUI documentation)
5. Then ask the model "What does 47867 represent?" and the model still remembers "it represents a bug"

## Logs

**Browser Console Logs:**
[OpenWebUI-1724846765350.log](https://github.com/user-attachments/files/16783011/OpenWebUI-1724846765350.log)

## Additional Information

I have also found that many people on some popular forums are encountering this issue, and it is not just in these few versions. The Context Length are ineffective, causing people to spend far more on API costs than expected. I hope it can be fixed as soon as possible!

**Comment by CookSleep** [comment]
Even if the Context Length is set to 50 tokens in the account settings, the problem still exists!

**Comment by tjbck** [comment]
Have you tried isolating this issue to Ollama?

**Reply by CookSleep** [reply]
Uh... but I have turned off any settings of Ollama (including but not limited to external connections), and I have not made any changes to the Ollama-related settings in the default settings

**Reply by sqiangcao99** [reply]
I have the same problem, has this problem been solved?

**Reply by mxdlzg** [reply]
Same problem. 
The pure api mode I use. Just as the front end of openai.

**Reply by Gilligan8** [reply]
I too am having the problem and I'm using the API to OpenAI, I mostly am sending to the gpt-4o-mini if that matters.

This was happening on 0.3.35 and still happening on 0.4.2

**Reply by zerofata** [reply]
Same issue running Open WebUI from a docker container. It ignores context length set from chat control or from advanced settings under the model and tries to send the whole chat to my vLLM backend using OpenAI connection causing it to refuse as the context is larger than what it's configured to accept. My chats have images in them.
![image](https://github.com/user-attachments/assets/f8446fd6-780e-4edb-a079-62ed64f48ea9)

**Comment by Gilligan8** [comment]
Honestly, just curious is this being looked at?

Seems like a pretty big bug, but I can't tell if it is being addressed.

**Reply by tjbck** [reply]
It's not a bug. Context length parameter exists only for Ollama, and NOT OpenAI chat completions.

As you can see, OpenAI chat completion endpoint doesn't support context length param: https://platform.openai.com/docs/api-reference/chat/create

I can definitely confirm num_ctx is being sent to Ollama:
![image](https://github.com/user-attachments/assets/eb6012d6-59be-4a64-a964-adbbb7211738)

**Reply by Gilligan8** [reply]
I could be wrong but I would think this would be a limit on how much open-webui is sending back on the request end.

Otherwise why am I hitting a limit?

The limit I'm hitting is that open-webui is trying to send openAI too much information.  I would think the first way to limit that would be at the open-webui point.

**Reply by smonux** [reply]
You may find this function helpful (haven't tried it, though)
https://openwebui.com/f/keke/context_length_filter
